src/main/java/edu/usf/cutr/tba/utils/StringUtils.java

```diff
@@ -84,4 +85,50 @@ public static String validateAndParseOutputPath(String newPath) {
         }
     }
 
+    /**
+     * If stringTimestamp is parsable to Long then return its long value, it it is not parsable, the function will find
+     * the best timeStamp option between a prefix value of the provided string or the MIN_VALUE of the Long class.
+     * @param timeStamp string containing a possible time stamp
+     * @return Long time stamp
+     */
+    public static long timeStampToLong(String timeStamp) {
+        if (!NumberUtils.isParsable(timeStamp)) {
+            // The timestamp must be a random string with a numeric prefix, get the prefix
+            int indexDash = timeStamp.indexOf("-");
+            if (indexDash == -1) {
+                // Dash not found, return MIN_VALUE
+                return Long.MIN_VALUE;
+            }
+            timeStamp = timeStamp.substring(0 , indexDash);
+            // if prefix is not Parsable, return MIN_VALUE
+            if (!NumberUtils.isParsable(timeStamp)) {
+                return Long.MIN_VALUE;
+            }
+        }
+        // timeStamp is parsable
+        return Long.parseLong(timeStamp);
+    }
+
+    /**
+     * If the provided timeStamp is parsable return the same string. If it is not parsable, the function will find the
+     * best timeStamp option between a prefix value of the provided string or the MIN_VALUE of the Long class.
+     * @param timeStamp string containing a possible time stamp
+     * @return string with a parsable time stamp
+     */
+    public static String parsableTimeStamp(String timeStamp) {
```

Comment:
I believe this method is unused and can be removed?

Reply:
Actually, looks like this is used too, but is almost a line-by-line duplicate of the above method (at first I thought this was a left-over first draft). Can you combine these methods to avoid duplicating this much code? Maybe passing in the entire `DeviceInformation` object and doing the null check within the method would help?